ecommerce/core/templatetags/core_extras.py

```diff
@@ -31,6 +36,90 @@ def settings_value(name):
     return getattr(settings, name)
 
 
+@register.simple_tag(name='is_special_coupon')
+def is_special_coupon(object_passed):
+    """
+    Checks if the coupon name meets the format criteria of a special coupon. The coupon name is
+    extracted from the 'object_passed' argument which could be an Order, Basket or OrderDiscount instance.
+
+    Arguments:
+        object_passed (Order or SimpleLazyObject or OrderDiscount): Order of the purchase, its basket or its OrderDiscount.
+
+    Returns:
+        bool: True if the coupon name is a valid special coupon, False otherwise.
+    """
+    is_order = isinstance(object_passed, Order)
+    is_basket = isinstance(object_passed, SimpleLazyObject)
+    is_order_discount = isinstance(object_passed, OrderDiscount)
+    coupon_name = ''
+
+    if not is_order and not is_basket and not is_order_discount:
+        return False
+
+    if is_order:
+        discount = object_passed.basket_discounts.first()
+        coupon_name = discount.voucher.name if discount else ''
+
+    if is_basket:
```

Comment:
Why don't you use if/else? @Jacatove

Reply:
I added a new function called `get_coupon_name` which its purpose is to retrieve the coupon name whatever Instance is provided, and I use if/elif/else statement within. @Squirrel18